elfi/methods/diagnostics.py

```diff
@@ -0,0 +1,260 @@
+
+    The rationale of the procedure is the following:
+    - Initially, the combinations of the to-be-assessed summary statistics is computed;
+    - In Stage 1, each summary-statistics combination is evaluated using the
+    minimum-entropy algorithm;
+    - In Stage 2, the minimum-entropy combination is selected,
+    and the `closest' datasets are identified;
+    - Further in Stage 2, for each summary-statistics combination,
+    the mean root sum of squared errors (MRSSE) is calculated over all `closest datasets',
+    and the minimum-MRSSE combination is chosen as the one with the optimal performance.
+
+    Attributes
+    ----------
+    list_ss : array_like
+        List of the to-be-assessed summary statistics.
+    simulator : elfi.Simulator
+        Simulator fit with the observations of the true/underlying model.
+    fn_distance : str or function
+        Function for calculating distance (can be a string if the function is implemented in ELFI).
+    seed : int, optional
+
+    References
+    ----------
+    [1] Nunes, M. A., & Balding, D. J. (2010).
+    On optimal selection of summary statistics for approximate Bayesian computation.
+    Statistical applications in genetics and molecular biology, 9(1).
+    [2] Blum, M. G., Nunes, M. A., Prangle, D., & Sisson, S. A. (2013).
+    A comparative review of dimension reduction methods in approximate Bayesian computation.
+    Statistical Science, 28(2), 189-208.
+
+    """
+
+    def __init__(self, list_ss, simulator, fn_distance, seed=0):
+        """Initialise the summary-statistics selection based on the Two-Stage Procedure.
+
+        Parameters
+        ----------
+        list_ss : array_like
+            List of the to-be-assessed summary statistics.
+        simulator : elfi.Simulator
+            Simulator fit with the observations of the true/underlying model.
+        fn_distance : str or function
+            Function for calculating distance
+            (can be a string if the function is implemented in ELFI).
+        seed : int, optional
+
+        """
+        self.list_ss = list_ss
+        self.simulator = simulator
+        self.fn_distance = fn_distance
+        self.seed = seed
+
+    def run(self, k, n_sim, n_acc, n_closest):
+        """Run the Two-Stage Procedure.
+
+        Parameters
+        ----------
+        k : int
+            Parameter for the kth-nearest-neighbour search.
+        n_sim : int
+            Number of the total ABC-rejection simulations.
+        n_acc : int
+            Number of the accepted ABC-rejection simulations.
+        n_closest : int
+            Number of the `closest' datasets.
+
+        Returns
+        -------
+        array_like
+            Set of the summary statistics showing the optimal performance.
+
+        """
+        # Obtain the combinations of the summary statistics.
+        sets_candidate_ss = self._create_ss_combinations(max_cardinality=k)
```

Comment:
This may be too limiting. Perhaps set `k` as the default, but give the user an optional argument to choose otherwise.

Reply:
Maybe (also) allow the use to give a list of combinations, in case he/she has some idea of which would be reasonable